src/components/UI/Drawer/components/OtherRecordsInSameLocation.tsx

```diff
@@ -26,14 +27,33 @@ type Props = {
 export const CloseByRecord = ({ drawerData }: Props) => {
   const { setDrawerData } = useMapActions();
   const { t } = useTranslation("home");
+  const router = useRouter();
+
   if (
     !drawerData ||
     !("closeByRecords" in drawerData) ||
     !drawerData.closeByRecords ||
     drawerData.closeByRecords.length <= 1
   )
     return null;
+
+  const updateStateID = (reference: number) => {
+    router.push(
+      {
+        pathname: "",
+        query: {
+          ...router.query, // list all the queries here
+          id: reference,
+        },
+      },
+      undefined,
+      {
+        shallow: true,
+      }
```

Comment:
what is all of these and why?

Reply:
to update the url after Switching between multiple locations